lambda/query-activity-log/tests/query-activity-log.test.ts

```diff
@@ -0,0 +1,292 @@
+  TxmaEvent: TEST_TXMA_EVENT,
+  ActivityLogEntry: activityLogEntry,
+};
+
+describe("queryActivityLog", () => {
+  beforeEach(() => {
+    dynamoMock.reset();
+
+    process.env.TABLE_NAME = tableName;
+
+    dynamoMock.on(QueryCommand).resolves({ Items: [activityLogEntry] });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("Query activity log", async () => {
+    const activityLog: ActivityLogEntry | undefined = await queryActivityLog(
+      userId,
+      sessionId
+    );
+    expect(activityLog).not.toBeNull();
+    expect(activityLog).toEqual(activityLogEntry);
+  });
+
+  test("Query user service empty list", async () => {
+    dynamoMock.on(QueryCommand).resolves({ Items: undefined });
+    const activityLog: ActivityLogEntry | undefined = await queryActivityLog(
+      userId,
+      sessionId
+    );
+    expect(activityLog).not.toBeNull();
+    expect(activityLog).toBeUndefined();
+  });
+});
+describe("validateTxmaEventBody", () => {
+  test("doesn't throw an error with valid txma data", () => {
+    expect(validateTxmaEventBody(TEST_TXMA_EVENT)).toBe(undefined);
+  });
+  test("throws error when client_id is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            timestamp: new Date().toISOString,
+            event_name: "AUTH_AUTH_CODE_ISSUED",
+            user: {
+              user_id: userId,
+            },
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test("throws error when timestamp is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            client_id: clientId,
+            event_name: "AUTH_AUTH_CODE_ISSUED",
+            user: {
+              user_id: userId,
+            },
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test("throws error when event name is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            client_id: clientId,
+            timestamp: new Date().toISOString,
+            user: {
+              user_id: userId,
+            },
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test(" throws error when user is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
+        services: [
+          {
+            client_id: clientId,
+            timestamp: new Date().toISOString,
+            event_name: "AUTH_AUTH_CODE_ISSUED",
+          },
+        ],
+      })
+    );
+    expect(() => {
+      validateTxmaEventBody(txmaEvent);
+    }).toThrowError();
+  });
+  test("throws error when user_id  is missing", () => {
+    const txmaEvent = JSON.parse(
+      JSON.stringify({
```

Comment:
This is ok but to be quicker, you can use object deconstruction and overrides to create tests parameters with altered fields 
Eg https://github.com/alphagov/di-account-management-backend/blob/main/lambda/save-raw-events/tests/save-raw-events.test.ts#L77


the syntax is 

`...<original object>, <field to mutate> : <new value>`


This means if the original object changes you don't need to re-write all the tests to update the object being created